### PR TITLE
Continue on AppInsights setup errors to simplify local development

### LINKED ISSUE #105: [Issue] Front-end Templates don't work if App Insights is not properly configured
See https://github.com/Azure/azure-dev/blob/main/templates/todo/web/react-fluent/src/services/telemetryService.ts#L29-L30.

If connection string is not set, the `loadApplicationInsights` triggers an exception that is not handled, and the front-end fails to load.

This mainly blocks developer from running the `web` app locally before provisioning Azure resources.

### PR DESCRIPTION
**Pull request by lmolkova**
Fixes #105
With this change if Application Insights setup throws (e.g. when running apps locally), we log the error but let an application work.

### CODE REVIEW

File: templates/todo/api/csharp-cosmos-sql/Program.cs [hunk 1]
```diff
@@ -15,8 +15,7 @@
     }
 }));
 builder.Services.AddControllers();
-var options = new ApplicationInsightsServiceOptions { ConnectionString = builder.Configuration["APPLICATIONINSIGHTS_CONNECTION_STRING"] };
-builder.Services.AddApplicationInsightsTelemetry(options);
+builder.Services.AddApplicationInsightsTelemetry(builder.Configuration);
```
Comment (lmolkova) [inline templates/todo/api/csharp-cosmos-sql/Program.cs:18]: .NET application Insights reads `APPLICATIONINSIGHTS_CONNECTION_STRING` env var from configuration/env on its own and supports binding to configuration, nothing extra is needed.

It also supports missing configuration and connection string writing to debug output or integrating with VS without ikey.

**Comment by lmolkova** [comment]
Tested Azure-Samples/todo-nodejs-mongo and Azure-Samples/todo-csharp-cosmos-sql locally and remotely, which should cover all changes. 

Python template already has a guard in place:

https://github.com/Azure/azure-dev/blob/6dec4c3e2bccdb9201e5320cec418964bc353ea0/templates/todo/api/python/todo/app.py#L38

File: templates/todo/web/react-fluent/src/services/telemetryService.ts [hunk 2]
```diff
@@ -27,16 +27,20 @@ export const getApplicationInsights = (): ApplicationInsights => {
...
+            if (!telemetry) {
+                return;
+            }
+            if (telemetry.tags) {
+                telemetry.tags['ai.cloud.role'] = "webui";
+            }
+        });
+    } catch(err) {
+        // TODO - proper logging for web
+        console.error("ApplicationInsights setup failed, ensure environment variable 'APPLICATIONINSIGHTS_CONNECTION_STRING' has been set.", err);
```
Comment (ellismg) [inline templates/todo/web/react-fluent/src/services/telemetryService.ts:42]: In the react app the value is read from `window.ENV_CONFIG.REACT_APP_APPLICATIONINSIGHTS_CONNECTION_STRING`.

**Review by ellismg — APPROVED** [review]
LGTM - I put a note about where I believe we source the key from for the SPA frontend.

**Review by wbreza — APPROVED** [review]
Thanks for the contributions.  One minor comments otherwise looks good to go :)

**Pull request merged**